### PR TITLE
NMS-14242: Add ROLE_CONFIG_EDITOR for editing config files from REST and UI

### PR DESCRIPTION
**Pull request by synqotik**
Add a new role, `ROLE_CONFIG_EDITOR` which allows access to view/edit configuration files via REST API and via the new UI. Only users with this role can view/editor these files via REST/UI.

This affects any REST API calls to `/rest/filesystem/**`. Even `ROLE_ADMIN` is denied access to view or edit files; however an admin can add the config editor role to themselves.

Changes include:

- updated Spring security configuration for the new role
- update Rest service code to disallow any but config editor role. May not be needed with spring XML configuration, but code was previously there an is an extra check
- update UI to only display "File Editor" navigation or allow routing for config editor users
- updates to documentation
- also a small update to DCB documentation for `retention-period` that I forgot on a previous PR

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14242

### CODE REVIEW

**Review by cgorantla — DISMISSED** [review]
LGTM!

My two cents :

With these changes, we are hiding the editing config files feature explicitly.
User may never know it exists.

File: docs/modules/operation/pages/device-config-backup/dcb-requisition.adoc [hunk 1]
```diff
@@ -20,6 +20,7 @@ If you want to enable device config backup on existing nodes in a requisition, s
 .. `password`: password for the device
 .. `script-file`: name of the device configuration script for this type of device
 .. `schedule`: a cron expression to specify the time to fetch the device backup configuration (default is `0 0 0 * * ?`, every 24 hrs at 12:00 a.m.)
+.. `retention-period`: an *optional* Java `Period` expression to specify the length of time before "stale" configuration data is deleted from the database (default is `P1Y`, one year)
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/dcb-requisition.adoc:23]:
```suggestion
.. `retention-period`: an optional Java `Period` expression to specify the length of time before "stale" configuration data is deleted from the database (default is `P1Y`, one year)
```

File: docs/modules/operation/pages/user-management/security-roles.adoc [hunk 2]
```diff
@@ -15,11 +15,14 @@ endif::opennms-prime[]
 | Description
 
 | ROLE_ADMIN*
-| Permissions to create, read, update, and delete in the web UI and the ReST API.
+| Permissions to create, read, update, and delete in the web UI and the ReST API (except see `ROLE_CONFIG_EDITOR` below).
```
Comment (Bonrob2) [inline docs/modules/operation/pages/user-management/security-roles.adoc:18]:
```suggestion
| Permissions to create, read, update, and delete in the web UI and the ReST API (except see ROLE_CONFIG_EDITOR below).
```

File: docs/modules/operation/pages/user-management/security-roles.adoc [hunk 3]
```diff
@@ -15,11 +15,14 @@ endif::opennms-prime[]
...
 
 | ROLE_ASSET_EDITOR
 | Permissions only to update the asset records from nodes.
 
+| ROLE_CONFIG_EDITOR
+| Permissions only to view and update configuration data via the REST API (and consequently, from the UI Preview). Note that even `ROLE_ADMIN` cannot view or edit configurations unless they also have the `ROLE_CONFIG_EDITOR` role. Also, for a user with `ROLE_CONFIG_EDITOR` to use the UI, they will also need the `ROLE_USER` or similar role.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/user-management/security-roles.adoc:24]:
```suggestion
| Permissions only to view and update configuration data via the REST API (and consequently, from the UI Preview). 
Note that even ROLE_ADMIN cannot view or edit configurations unless they also have the ROLE_CONFIG_EDITOR role. 
Also, for a user with ROLE_CONFIG_EDITOR to use the UI, they will also need the ROLE_USER or similar role.
```

**Comment by synqotik** [comment]
> With these changes, we are hiding the editing config files feature explicitly. User may never know it exists.

Agreed, this is the implemented behavior. Will doublecheck that it's the desired behavior.

**Review by Bonrob2 — COMMENTED** [review]
Docs look good, a few minor formatting changes.

File: docs/modules/operation/pages/device-config-backup/dcb-requisition.adoc (hunk 1 above)
Comment (synqotik) [reply docs/modules/operation/pages/device-config-backup/dcb-requisition.adoc]: Fixed

File: docs/modules/operation/pages/user-management/security-roles.adoc (hunk 2 above)
Comment (synqotik) [reply docs/modules/operation/pages/user-management/security-roles.adoc]: Fixed

File: docs/modules/operation/pages/user-management/security-roles.adoc (hunk 3 above)
Comment (synqotik) [reply docs/modules/operation/pages/user-management/security-roles.adoc]: Fixed

**Pull request merged**